File: server/java/src/main/java/com/onlinepayments/example/service/interfaces/clients/PaymentLinkClientInterface.java
package com.onlinepayments.example.service.interfaces.clients;

import com.onlinepayments.example.service.dtos.paymentlinks.CreatePaymentLinkRequestDto;
import com.onlinepayments.example.service.dtos.paymentlinks.CreatePaymentLinkResponseDto;

public interface PaymentLinkClientInterface {
    CreatePaymentLinkResponseDto createPaymentLink(CreatePaymentLinkRequestDto request);
}

File: server/nodejs/src/infrastructure/interfaces/payment-link-client-interface.ts
import { FastifyBaseLogger } from 'fastify';
import { CreatePaymentLinkRequestDto } from '../../business/dtos/payment-link/create-payment-link-request-dto';
import { CreatePaymentLinkResponseDto } from '../../business/dtos/payment-link/create-payment-link-response-dto';

export interface IPaymentLinkClient {
    createPaymentLink(
        request: CreatePaymentLinkRequestDto,
        logger: FastifyBaseLogger,
    ): Promise<CreatePaymentLinkResponseDto>;
}
